### PR TITLE
automate addition of extra mission/unit if delegate controls respecti…

### LINKED ISSUE #43: Ošetření misí získatelných před neutrály
Nyní je systém rigidně nastaven, že každá frakce má právě jednu misi za kolo a je jí vždy první akce.

Ve hře jsou ale možnosti (přes neutrální rody) získat mise navíc. Minule jsme je přidávali jako akce s názvem "mise", ale problém je, že nefunguje výběr misí, hráči ji musejí slovně popsat, což komplikuje jak zadávání, tak vyhodnocení.

Ideálně chceme u každé "akce" možnost nastavit, zda se jedná o misi nebo ne. Pokud se jedná o misi, tak uživatel může vybírat z přednastaveného seznamu misí.

### PR DESCRIPTION
**Pull request by Snow-rider**
if delegate controls respective neutral house(s)

extra mission for each of the following neutral houses owned:
- Riches
- Samarama

extra unit action for each of the following neutral houses owned:
- Genissai
- Yuzovka

closes #43

### CODE REVIEW

File: common/config.ts [hunk 1]
```diff
@@ -112,3 +112,7 @@ export function findValueName(rows: ValueName[], value: string) {
     if (row == undefined) return "N/A"
     return row.name
 }
+
+export const EXTRA_MISSION_HOUSES: string[] = ["Riches", "Samarama"]
```
Comment (davidvavra) [inline common/config.ts:116]: It would be better to add some boolean flags to the votingRight and not match according to name.

Comment (Snow-rider) [reply common/config.ts:116]: Hmm ok so when we assign the house to the delegate, then add flags. I will check it out.

Comment (Snow-rider) [reply common/config.ts:116]: But even when we add the flags at the assignment of who controls the neutral house, we would need to match based on the name.

Comment (Snow-rider) [reply common/config.ts]: Fixed: extraMission or extraUnit is added for voting right in the landsraad ui, and saved as DB variable for votingRight and then read from there when generating actions.

**Comment by Snow-rider** [comment]
Tested that BV for extra mission & extra unit is subtracted in order from the total BV and both actions are exported in the csv (together with normal missions/unit actions) when actions are exported.

When this is deployed to prod, we just need to set up the "extraMission" and "extraUnit" in the prod DB using the landsraad UI.

File: swiss/src/app/round/round.component.ts [hunk 2]
```diff
@@ -277,15 +280,22 @@ export class RoundComponent implements OnInit {
...
           combineLatest(
             this.db.list("delegateRounds").snapshotChanges(),
             this.db.list("units").snapshotChanges(),
-            (delegateRounds, units) => {
-              return { delegateRounds: delegateRounds, units: units }
+            this.db.list("landsraad/votingRights").valueChanges(),
+            (delegateRounds, units, votingRights) => {
+              // we only need the information for the special houses which add unit or mission
+              let votingRightsFiltered = votingRights.filter((votingRight: VotingRight) => {
+                return votingRight.extraMission.toLowerCase() === "true" || votingRight.extraUnit.toLowerCase() === "true"
```
Comment (Snow-rider) [inline swiss/src/app/round/round.component.ts:296]: even though we try to save boolean values, firebase somehow stores strings. I read about it and it seems other users also hit this problem as it is feature/bug of firebase. So we have to do string comparison.

**Review by davidvavra — APPROVED** [review]

**Pull request merged**